plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2374,78 +2306,36 @@ bool producer_plugin_impl::process_unapplied_trxs(const fc::time_point& deadline
-      if (sch_itr->delay_until > pending_block_time)
-         break; // not scheduled yet
-      if (exhausted || deadline <= fc::time_point::now()) {
-         exhausted = true;
-         break;
-      }
-      if (sch_itr->published >= pending_block_time) {
-         ++sch_itr;
-         continue; // do not allow schedule and execute in same block
-      }
+   const auto&        expired_idx         = chain.db().get_index<generated_transaction_multi_index, by_expiration>();
+   const auto expired_size                = expired_idx.size();
+   auto               expired_itr         = expired_idx.begin();
 
-      if (blacklist_by_id.find(sch_itr->trx_id) != blacklist_by_id.end()) {
-         ++sch_itr;
-         continue;
-      }
-
-      const transaction_id_type trx_id         = sch_itr->trx_id; // make copy since reference could be invalidated
-      const auto                sch_expiration = sch_itr->expiration;
-      auto                      sch_itr_next   = sch_itr; // save off next since sch_itr may be invalidated by loop
-      ++sch_itr_next;
-      const auto next_delay_until = sch_itr_next != sch_idx.end() ? sch_itr_next->delay_until : sch_itr->delay_until;
-      const auto next_id          = sch_itr_next != sch_idx.end() ? sch_itr_next->id : sch_itr->id;
-
-      num_processed++;
-
-      // configurable ratio of incoming txns vs deferred txns
-      while (incoming_trx_weight >= 1.0 && itr != end) {
-         if (deadline <= fc::time_point::now()) {
-            exhausted = true;
-            break;
-         }
-
-         incoming_trx_weight -= 1.0;
-
-         auto trx_meta = itr->trx_meta;
-         bool api_trx  = itr->trx_type == trx_enum_type::incoming_api;
-
-         auto trx_tracker = _time_tracker.start_trx(trx_meta->is_transient());
-         push_result pr = push_transaction(deadline, trx_meta, api_trx, itr->return_failure_trace, trx_tracker, itr->next);
-
-         exhausted = pr.block_exhausted;
-         if (pr.trx_exhausted) {
-            ++itr; // leave in incoming
-         } else {
-            itr = _unapplied_transactions.erase(itr);
-         }
-
-         if (exhausted)
-            break;
+   while (expired_itr != expired_idx.end()) {
+      if (expired_itr->expiration >= pending_block_time) {
+         break; // not expired yet
       }
 
       if (exhausted || deadline <= fc::time_point::now()) {
          exhausted = true;
          break;
       }
 
+      const transaction_id_type trx_id             = expired_itr->trx_id; // make copy since reference could be invalidated
+      auto                      expired_itr_next   = expired_itr; // save off next since expired_itr may be invalidated by loop
+      ++expired_itr_next;
+      const auto next_expired = expired_itr_next != expired_idx.end() ? expired_itr_next->expiration : expired_itr->expiration;
+      const auto next_id      = expired_itr_next != expired_idx.end() ? expired_itr_next->id : expired_itr->id;
+
+      num_processed++;
+
```

Comment:
`max_trx_time` below can be hard-coded to `fc::microseconds::maximum()` as it is not used in `push_scheduled_transaction` when trx is expired.